### PR TITLE
PDS Longevity Test - RebootNodes

### PR DESCRIPTION
**Pull request by Madanagopal19**
This PR has the changes to use the existing longevity tests to run with pds apps

regression_logs: https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds-branch-build-test/362/console

### CODE REVIEW

File: drivers/pds/dataservice/dataservices.go [hunk 1]
```diff
@@ -382,6 +387,12 @@ func (d *DataserviceType) CreateSchedulerContextForPDSApps(pdsApps []*pds.Models
 	var ctx *scheduler.Context
 
 	for _, dep := range pdsApps {
+		dep.Namespace, err = components.Namespace.GetNamespace(*dep.NamespaceId)
+		if err != nil {
+			log.Errorf("error while getting namespace")
```
Comment (lsrinivas-pure) [inline drivers/pds/dataservice/dataservices.go:392]: return the error here

File: drivers/scheduler/k8s/k8s.go [hunk 2]
```diff
@@ -3604,6 +3670,26 @@ func (k *K8s) ValidateVolumes(ctx *scheduler.Context, timeout, retryInterval tim
...
+			ss, err := k8sApps.GetStatefulSet(obj.GetClusterResourceName(), *obj.Namespace.Name)
+			if err != nil {
+				return &scheduler.ErrFailedToValidateStorage{
+					App:   ctx.App,
+					Cause: fmt.Sprintf("Failed to get StatefulSet: %v , Namespace: %v. Err: %v", obj.Name, obj.Namespace, err),
+				}
+			}
+			// Providing the scaling factor in timeout
+			scalingFactor := *ss.Spec.Replicas
+			if *ss.Spec.Replicas > *ss.Spec.Replicas {
```
Comment (lsrinivas-pure) [inline drivers/scheduler/k8s/k8s.go:3683]: can you check here seems u r  comparing same things

File: drivers/scheduler/k8s/k8s.go [hunk 3]
```diff
@@ -3886,6 +3972,35 @@ func (k *K8s) GetVolumes(ctx *scheduler.Context) ([]*volume.Volume, error) {
 				vol.Annotations[key] = val
 			}
 			vols = append(vols, vol)
+		} else if pdsobj, ok := specObj.(*pds.ModelsDeployment); ok {
+			ss, err := k8sApps.GetStatefulSet(*pdsobj.ClusterResourceName, *pdsobj.Namespace.Name)
+			if err != nil {
+				log.Fatalf("Failed to get statefulset pods %v", err)
```
Comment (lsrinivas-pure) [inline drivers/scheduler/k8s/k8s.go:3978]: do not log fatal just return error

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: drivers/pds/dataservice/dataservices.go (hunk 1 above)
Comment (Madanagopal19) [reply drivers/pds/dataservice/dataservices.go]: done

File: drivers/scheduler/k8s/k8s.go (hunk 2 above)
Comment (Madanagopal19) [reply drivers/scheduler/k8s/k8s.go]: done

File: drivers/scheduler/k8s/k8s.go (hunk 3 above)
Comment (Madanagopal19) [reply drivers/scheduler/k8s/k8s.go]: done

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by aravindranath14 — APPROVED** [review]
Looks Good!!

**Pull request merged**